docs/private_clusters.md

```diff
@@ -0,0 +1,42 @@
+# Creating Private GKE Clusters
+
+To create a private GKE cluster, you can use one of the [private submodules](../modules).
+
+Note that a private cluster is inherently more restricted and greater care must be taken in configuring networking ingress/egress.
+
+## Private Cluster Endpoints
+When creating a [private cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters), nodes are provisioned with private IPs.
+The Kubernetes master endpoint is also [locked down](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#access_to_the_cluster_endpoints), which affects these module features:
+- `configure_ip_masq`
+- `stub_domains`
```

Comment:
Can we mention the `workload-identity` module here or should we add a small blurb in that module?

Reply:
I'd say add a blurb in that module (and possibly link to this doc).